Automattic/pocket-casts-ios
# [Up Next Shuffle] Enable the Feature for Plus and Patron only

| 📘 Part of: [Up Next Shuffle](https://github.com/orgs/Automattic/projects/1179/views/1)
|:---:|

This PR adds:
• The feature available for Plus and Patron user only
• Hides the shuffle button and clear button if the queue is empty
• Update Changelog

| Plus Icon | Default Icon | Empty Screen |
| -------- | ------- | ------- |
| ![RocketSim_Screenshot_iPhone_16_6 1_2024-10-31_14 37 54](https://github.com/user-attachments/assets/10018c98-46cc-4594-b34b-83ea7fc28454) | ![RocketSim_Screenshot_iPhone_16_6 1_2024-10-31_14 38 19](https://github.com/user-attachments/assets/d5f053e3-7c17-471d-b0ae-720ce66f9c6e) | ![RocketSim_Screenshot_iPhone_16_6 1_2024-10-31_14 38 36](https://github.com/user-attachments/assets/10ae68ff-57cf-4ef1-b303-1b4e2e0d6e8e) |

## To test

- CI must be 🟢 
- Run the app with the StoreKit scheme or run this PR on a real device with a sandbox account
- Run a fresh install and log in with a non plus/patron user
- Add few episodes in the queue
- Go to Up Next 
- Confirm the Shuffle button gold is present
- Clear the queue
- Confirm both shuffle and clear button disappear from the view
- Add few episodes again
- Go back to Up Next
- Tap shuffle gold
- Confirm the paywall is presented
- Purchase a plan
- When the paywall is dismissed confirm the Shuffle icon is the gray one
- Confirm tapping the shuffle doesn't promtp the paywall and enable/disable the feature

## Checklist
